### ISSUE TITLE
Distribution of NHI is very different than data

### ISSUE THREAD

**Issue by londumas** [igmhub/LyaCoLoRe#26]
In the following plot I compare the distribution of NHI in data (DR14_DLA_Noterdaeme) vs. the catalog in the mocks. We can see that the distribution is very different.
I don't know much about DLA, it could also be an issue with the efficiency in detecting DLA in data.

![nh1_distribution](https://user-images.githubusercontent.com/3166436/47574866-dd8b0080-d8fd-11e8-8b35-cde61f24c4a3.png)

**Comment by fjaviersanchez** [comment]
Ok, I fixed a bug in the code that was pushing the distribution towards larger NHI and to be nearly flat. I tested the fix for both solutions with `pyigm` and without:

Without `pyigm`:

![image](https://user-images.githubusercontent.com/13971732/47714463-3f6ca280-dc3d-11e8-9c33-85da0959c7ea.png)


With `pyigm`:

![image](https://user-images.githubusercontent.com/13971732/47714497-54493600-dc3d-11e8-8307-1d96d13aab86.png)

I am opening a PR to solve this as well.

**Comment by andreufont** [comment]
Hi @jfarr03 - can you generate a v4.1 from the same CoLoRe box than v4.0.0, but with this DLA fix? It would be good to run the full notebook of quality assessment before closing this issue.

**Comment by jfarr03** [comment]
Will do!

**Comment by londumas** [comment]
@jfarr03, please tell me when it is ready and I will have a look at it.

**Comment by jfarr03** [comment]
Yeah I will do - currently waiting on a bug fix but then will run everything through.

**Comment by jfarr03** [comment]
Sorry for the delay - there's a new lot of transmission files at /project/projectdirs/desi/mocks/lya_forest/london/v4.1.0 with the new NHI fixes included now!

**Comment by andreufont** [comment]
@jfarr03 - should I run quickquasars on it? Also, could you add a short README and documentation in the wiki page?

**Comment by londumas** [comment]
@andreufont, let me first look at the NHI distribution. If it is ok, then let's run quickquasar.

**Comment by londumas** [comment]
@jfarr03, I can't find the catalog of DLA in `/project/projectdirs/desi/mocks/lya_forest/london/v4.1.0/`. Should I still produce it? Or is it hidden somewhere?

**Comment by londumas** [comment]
@fjaviersanchez, @jfarr03, @andreufont,
The version `v4.1.0` is better but still not perfect. Could you have a look at what is happening?

![nhi_distribution](https://user-images.githubusercontent.com/3166436/48218712-babe0a80-e347-11e8-84dd-7d08a66270c3.png)

![nhi_distribution_2](https://user-images.githubusercontent.com/3166436/48218714-babe0a80-e347-11e8-8f78-936b5847234b.png)

**Comment by fjaviersanchez** [comment]
Doing so now. I am also checking the bias problems. Let's see if I find something

**Comment by jfarr03** [comment]
@londumas  The DLA_master file is being made now! There's an initial one there with basic data in, but I'm making a proper one now with the details we discussed at the meeting (RA,DEC,Z_QSO,... etc)

**Comment by londumas** [comment]
@jfarr03, perfect, tell me when I can test it.

**Comment by fjaviersanchez** [comment]
Ok, I found out what was going on. I was interpreting log10 instead of log... Now the agreement is perfect:

![image](https://user-images.githubusercontent.com/13971732/48292754-87a37600-e430-11e8-896e-3d4c050fe479.png)

I am doing a PR with these changes.

**Comment by londumas** [comment]
@fjaviersanchez, @jfarr03, good. Tell me when I can test the results in the next version.

**Comment by fjaviersanchez** [comment]
@londumas I want to figure out the bias fix before generating another table but I'll give you the heads up. Thanks a lot!

**Comment by andreufont** [comment]
It looks good @fjaviersanchez . Should we close this issue, then?

**Comment by londumas** [comment]
@andreufont and @fjaviersanchez, let's test it on the next run of the mocks, run the desi_QA and then close.

**Comment by jfarr03** [comment]
Glad this issue got fixed! I've run LyaCoLoRe again with the fixed DLA method, and with new tuning of the small scale power (shouldn't affect NHI distribution). The files are just transferring to a new v4.2.0 folder.

**Comment by londumas** [comment]
@jfarr03, Thank you very much. Some remarks: in `master_DLA.fits` all `Z_DLA_RSD = 0` and `N_HI_DLA` (or `NHI`, as you want) is missing.

**Comment by londumas** [comment]
@jfarr03, and in the transmission file `DZ_DLA_RSD` has a `D` as prefix. Maybe this explains the `Z_DLA_RSD = 0`.
Could I know why you recall inside the `DLA` hdu of the transmission files the `RA, DEC, Z_QSO_NO_RSD, Z_QSO_RSD`? Isn't that repeating information from the first hdu?
I totally agree that we need these information in the `master_DLA.fits`, but do we really need them twice in the transmission file?

**Comment by jfarr03** [comment]
Good spots - I'll fix these now. It's not entirely necessary, I agree, but it makes things much faster when building the master as otherwise you have to search

**Comment by londumas** [comment]
@jfarr03, My code here, under `Create a catalog` is not that long: https://github.com/igmhub/desi_QA/blob/master/Mocks/QA_DLA.ipynb. I think we can leave with it. But I leave you decide. Do you know how much extra space it takes to keep all of these?

**Comment by jfarr03** [comment]
Not sure how much extra space it is - I can't imagine it's much in relation to the skewers data? Still, it's worth trying to optimise a bit - I'll have a quick look at it now.

**Comment by londumas** [comment]
Apart from these, it looks good for the NHI distribution, thanks @fjaviersanchez and @jfarr03.
@andreufont, @alxogm, @ngbusca, do we want something even more closer to the data, or this is good enough? Keep in mind that in the data there is the effect of efficiency of detection, bringing some uncertainties in the blue points.
Maybe we should work a bit on the high NHI tail of the distribution, at NHI=22.5 we have a one dex difference.

![nhi_distribution](https://user-images.githubusercontent.com/3166436/48565200-39b9c280-e8b5-11e8-8025-6a54acefd8d4.png)

![log_distribution_nhi](https://user-images.githubusercontent.com/3166436/48565606-2b1fdb00-e8b6-11e8-92e2-5239be97a962.png)

![nhi_probability](https://user-images.githubusercontent.com/3166436/48565201-3a525900-e8b5-11e8-8456-580d8c4a41de.png)

**Comment by jfarr03** [comment]
@londumas there's a new DLA master in v4.2.0 now, and future versions will do away with the unnecessary data in the transmission files too! And thanks for making the plots, glad they're looking better

**Comment by londumas** [comment]
@jfarr03, Thank you for the support. There is a bug in `Z_DLA_RSD`, that is not in `Z_DLA_NO_RSD`.
Maybe a `1+z`?
![distribution_zdla](https://user-images.githubusercontent.com/3166436/48569273-763dec00-e8be-11e8-89ab-78d0f0d0c059.png)

**Comment by jfarr03** [comment]
Ah yes that was just a type confusing DZ_DLA_RSD and Z_DLA_RSD. Fixed it and reproducing the file now

**Comment by jfarr03** [comment]
@londumas it should be fixed now!
